File: src/app/model/Notification.java
package app.model;

import java.util.Date;

public class Notification {
  public int byUserId;
  public Date time;

  private Notification() {
  }

  public static class BirthdayNotification extends Notification {
  }

  public static class LikeArticleNotification extends Notification {
    public int articleId;
  }

  public static class LikeCommentNotification extends Notification {
    public int commentId;
  }

  public static class CommentNotification extends Notification {
    public int commentId;
  }

  public static class ReplyCommentNotification extends Notification {
    public int commentId;
  }

  public static class SkillEndorseNotification extends Notification {
    public int skillId;
  }

  public static class ProfileVisitNotification extends Notification {
  }
}

File: src/app/model/Notification.kt
package app.model

import java.util.Date

sealed class Notification private constructor() {
	open var byUserId: Int = 0
	open var time: Date = Date(0)

	data class BirthdayNotification(
		override var byUserId: Int,
		override var time: Date,
	) : Notification()

	data class ProfileVisitNotification(
		override var byUserId: Int,
		override var time: Date,
	) : Notification()

	data class LikeArticleNotification(
		override var byUserId: Int,
		override var time: Date,
		var articleId: Int,
	) : Notification()

	data class LikeCommentNotification(
		override var byUserId: Int,
		override var time: Date,
		var commentId: Int,
	) : Notification()

	data class CommentNotification(
		override var byUserId: Int,
		override var time: Date,
		var commentId: Int,
	) : Notification()

	data class ReplyCommentNotification(
		override var byUserId: Int,
		override var time: Date,
		var commentId: Int,
	) : Notification()

	data class SkillEndorseNotification(
		override var byUserId: Int,
		override var time: Date,
		var skillId: Int,
	) : Notification()
}
